src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -33,7 +33,10 @@
             r"(-- |#)[^\n]*",
             CommentSegment,
             segment_kwargs={"trim_start": ("-- ", "#")},
-        )
+        ),
+        RegexLexer(
+            "single_quote", r"(?s)('')+?(?!')|('.*?(?<!')(?:'')*'(?!'))", CodeSegment
+        ),
```

Comment:
This just updates the quoted literal in mysql to match postgres. N.B. this matches the individual quoted literals, not the concatenated combination (for reason mentioned in PR description).